src/v/config/broker_authn_endpoint.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "config/broker_authn_endpoint.h"
+
+#include "kafka/client/exceptions.h"
+#include "model/metadata.h"
+#include "utils/string_switch.h"
+
+namespace config {
+
+std::string_view to_string_view(broker_authn_method m) {
+    switch (m) {
+    case broker_authn_method::none:
+        return "none";
+    case broker_authn_method::sasl:
+        return "sasl";
+    case broker_authn_method::mtls_identity:
+        return "mtls_identity";
+    }
+}
+
+template<>
+std::optional<broker_authn_method>
+from_string_view<broker_authn_method>(std::string_view sv) {
+    return string_switch<broker_authn_method>(sv)
+      .match("none", broker_authn_method::none)
+      .match("sasl", broker_authn_method::sasl)
+      .match("mtls_identity", broker_authn_method::mtls_identity)
+      .default_match(broker_authn_method::none);
```

Comment:
nit: seems like since you have a default match using none that the return optional type is never nullopt